File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/repository/UserRepository.kt
package com.sudo248.sudoo.domain.repository

import com.sudo248.base_android.core.DataState
import com.sudo248.sudoo.domain.entity.user.AddressSuggestion
import com.sudo248.sudoo.domain.entity.user.User

interface UserRepository {
    suspend fun getUserInfo(): DataState<User, Exception>
    suspend fun updateUser(user: User, isUpdate: Boolean = false): DataState<User, Exception>
    suspend fun getAddressSuggestionProvince(): DataState<List<AddressSuggestion>, Exception>
    suspend fun getAddressSuggestionDistrict(provinceId: Int): DataState<List<AddressSuggestion>, Exception>
    suspend fun getAddressSuggestionWard(districtId: Int): DataState<List<AddressSuggestion>, Exception>
}

File: client/web/sudoo/lib/domain/repository/user_repository.dart
import 'dart:core';

import 'package:sudoo/domain/model/user/user.dart';

import '../core/data_state.dart';
import '../model/user/address_suggestion.dart';

abstract class UserRepository {
  Future<DataState<User, Exception>> getUser();
  Future<DataState<User, Exception>> updateUser(User user);

  Future<DataState<List<AddressSuggestion>, Exception>> getSuggestionProvince();
  Future<DataState<List<AddressSuggestion>, Exception>> getSuggestionDistrict(int provinceId);
  Future<DataState<List<AddressSuggestion>, Exception>> getSuggestionWard(int districtId);

}